source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,196 @@
+                    }
+                }
+
+            }
+            fe_face_values.reinit(cell, face_idx);
+
+            if (at_upper_surface)
+              {
+                in.reinit(fe_face_values,
+                          cell,
+                          this->introspection(),
+                          this->get_solution());
+                //in.requested_properties = MaterialModel::MaterialProperties::viscosity;
+
+                this->get_material_model().fill_additional_material_model_inputs(in,
+                                                                                 this->get_solution(),
+                                                                                 fe_face_values,
+                                                                                 this->introspection());
+
+                this->get_material_model().evaluate(in, out);
+
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    // Calculate the square root of the second moment invariant for the deviatoric strain rate tensor.
+                    //edot_ii = std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[q])))),
+                    //          min_strain_rate);
+
+                    const double local_second_invariant_of_strain_rate = (std::max(std::sqrt(std::fabs(second_invariant(deviator(in.strain_rate[q])))),min_strain_rate) * fe_face_values.JxW(q));
+                    const double local_surface_area = fe_face_values.JxW(q);
+
+                    // second invariant of strain rate over the whole surface
+                    local_second_invariant_of_strain_rate_integral += local_second_invariant_of_strain_rate;
+                    local_surface_area_integral += local_surface_area;
+                  }
+              }
+          }
+
+      double total_second_invariant_of_strain_rate (local_second_invariant_of_strain_rate_integral.size());
+      Utilities::MPI::sum (local_second_invariant_of_strain_rate_integral, this->get_mpi_communicator(), total_second_invariant_of_strain_rate);
+
+      double total_surface_area (local_surface_area_integral.size());
+      Utilities::MPI::sum (local_surface_area_integral, this->get_mpi_communicator(), total_surface_area);
+
+
+      //Merge so column 1 is strain rate invariants and column 2 are the corresponding areas
+      for (int i = 0; i < local_second_invariant_of_strain_rate_integral.size(); ++i)
+        {
+          local_second_invariant_of_strain_rate_and_area[i][0] = local_second_invariant_of_strain_rate_and_area[i];
+          local_second_invariant_of_strain_rate_and_area[i][1] = local_surface_area_integral[i];
+        }
+
+      //sort local second invariants of strain rate in size order
+      sorted_local_second_invariant_of_strain_rate = sort(local_second_invariant_of_strain_rate[:][0].begin(), local_second_invariant_of_strain_rate[:][0].end(), greater<int>());
```

Comment:
`vector[:]` is syntax from Python, but not valid C++. What do you try to do? If you want to sort `local_second_invariant_of_strain_rate_and_area` it should look like this: 
```
```suggestion
      std::sort(local_second_invariant_of_strain_rate.begin(), local_second_invariant_of_strain_rate.end());
```
However you want to keep the strain rate and area of each cell associated, correct? Then I would suggest you store them in a data structure of type `std::vector<std::pair<double,double>>`. std::sort should understand automatically that it should sort by the first of the two pair members (see https://www.geeksforgeeks.org/sorting-vector-of-pairs-in-c-set-1-sort-by-first-and-second/).

Also this is the part of the algorithm that will not work in parallel unless you communicate every vector entry from every MPI process with one MPI process (e.g. process 0). Your `MPI::sum` command above does not do that, you would need to use an `MPI::gather`. For now let's ignore this part and make the algorithm work in serial computations.

Reply:
I see, thanks for the explanation.